NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -84,8 +87,21 @@ - (void)viewDidLoad {
 
 - (void)updateUI {
     self.collectionView.hidden = self.viewModel.collectionViewHidden;
+    self.errorMessageLabel.text = self.viewModel.errorMessageLabelText;
     self.errorMessageLabel.hidden = self.viewModel.errorMessageLabelHidden;
-    [self.collectionView reloadData];
+//    [self.collectionView reloadData];
+    
+}
+
+- (void)insertItemsFrom:(NSUInteger)startIndex To:(NSUInteger)endIndex {
+    [self.collectionView performBatchUpdates:^{
+        NSMutableArray* array = [[NSMutableArray alloc] init];
+        for (NSUInteger i = startIndex; i <= endIndex; i++) {
+            [array addObject: [NSIndexPath indexPathForItem:i inSection:0]];
+        }
```

Comment:
In case there's an error, you've called this line: `self.nextArticlesCallback(0, 0, error);`
In combination with the above line, even if there's an error, a new item is still added to the `collection view`. The app will crash because there will be a mismatch between the number of items in the data source and the number of items in the `collection view`.